python-sdk/nuscenes/eval/detection/main.py

```diff
@@ -119,58 +127,86 @@ def run(self) -> DetectionMetrics:
+
+        for detection_name in self.cfg.class_names:
+            class_pr_curve(md_list, metrics, detection_name, self.cfg.min_precision, self.cfg.min_recall,
+                           savepath=savepath(detection_name+'_pr'))
+
+            class_tp_curve(md_list, metrics, detection_name, self.cfg.min_recall, self.cfg.dist_th_tp,
+                           savepath=savepath(detection_name+'_tp'))
+
+        for dist_th in self.cfg.dist_ths:
+            dist_pr_curve(md_list, metrics, dist_th, self.cfg.min_precision, self.cfg.min_recall,
+                          savepath=savepath('dist_pr_'+str(dist_th)))
+
+
+def main():
+    args = parser.parse_args()
+    result_path = os.path.expanduser(args.result_path)
+    output_dir = os.path.expanduser(args.output_dir)
+    eval_set = args.eval_set
+    dataroot = args.dataroot
+    version = args.version
+    verbose = bool(args.verbose)
 
-if __name__ == "__main__":
-    pass
-    # Settings.
-    # parser = argparse.ArgumentParser(description='Evaluate nuScenes result submission.',
-    #                                  formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    # parser.add_argument('--result_path', type=str, default='~/nuscenes-metrics/results.json',
-    #                     help='The submission as a JSON file.')
-    # parser.add_argument('--output_dir', type=str, default='~/nuscenes-metrics',
-    #                     help='Folder to store result metrics, graphs and example visualizations.')
-    # parser.add_argument('--eval_set', type=str, default='val',
-    #                     help='Which dataset split to evaluate on, e.g. train or val.')
-    # parser.add_argument('--dataroot', type=str, default='/data/sets/nuscenes',
-    #                     help='Default nuScenes data directory.')
-    # parser.add_argument('--version', type=str, default='v0.5',
-    #                     help='Which version of the nuScenes dataset to evaluate on, e.g. v0.5.')
-    # parser.add_argument('--plot_examples', type=int, default=0,
-    #                     help='Whether to plot example visualizations to disk.')
-    # parser.add_argument('--verbose', type=int, default=1,
-    #                     help='Whether to print to stdout.')
-    # args = parser.parse_args()
-    # result_path = os.path.expanduser(args.result_path)
-    # output_dir = os.path.expanduser(args.output_dir)
-    # eval_set = args.eval_set
-    # dataroot = args.dataroot
-    # version = args.version
-    # eval_limit = args.eval_limit
-    # plot_examples = bool(args.plot_examples)
-    # verbose = bool(args.verbose)
-    #
     # Init.
-    # random.seed(43)
-    # nusc_ = NuScenes(version=version, verbose=verbose, dataroot=dataroot)
-    # nusc_eval = NuScenesEval(nusc_, result_path, eval_set=eval_set, output_dir=output_dir, verbose=verbose)
-    #
-    # Visualize samples.
+    random.seed(43)
+    cfg = config_factory('cvpr_2019')
```

Comment:
Take in config version as an arg. Verify that arg is valid for `config_factory`.